### PR TITLE
Set depsolver timeout based on chef_objects app config

### PR DESCRIPTION
**Pull request by seth**
If not provided, a default timeout of 2000 ms is used.

This depends on depsolver >= 1.1.0 for depsolver:solve/3.

### CODE REVIEW

File: src/chef_depsolver.erl [hunk 1]
```diff
@@ -108,3 +110,13 @@ sanitize_semver({ok, WithSemver}) ->
     {ok, XYZOnly};
 sanitize_semver(Error) ->
     Error.
+
+depsolver_timeout() ->
+    case application:get_env(chef_objects, depsolver_timeout) of
```
Comment (oferrigni) [inline src/chef_depsolver.erl:115]: Why not use envy here?

Comment (seth) [inline src/chef_depsolver.erl:115]: chef_objects does not depend on envy. I'm reluctant to introduce a dependency in general, but particularly as part of this patch. There are other uses of `application:get_env` which we'd want to convert in one go.

File: src/chef_depsolver.erl [hunk 2]
```diff
@@ -108,3 +110,13 @@ sanitize_semver({ok, WithSemver}) ->
...
+        undefined ->
+            ?DEFAULT_DEPSOLVER_TIMEOUT;
+        {ok, Value} when is_integer(Value) ->
+            Value;
+        Bad ->
+            error({invalid_config, {chef_objects, depsolver_timeout, Bad}})
```
Comment (oferrigni) [inline src/chef_depsolver.erl:121]: Do we need to log this error?

**Comment by oferrigni** [comment]
Even with the logging question, LGTM

File: src/chef_depsolver.erl (hunk 2 above)
Comment (seth) [inline src/chef_depsolver.erl:121]: I think the error will get logged as part of an ensuing crash.

**Pull request merged**